### ISSUE TITLE
MD013 cannot normalize italicized text followed by a number

### ISSUE THREAD

**Issue by denialhaag** [rvben/rumdl#760]
### rumdl version

0.2.43

### Installation method

Pre-built binary

### Bug description

This one is admittedly very niche, but MD013 fails to normalize the specific example below.

### Steps to reproduce

Run `rumdl check` on the following file:

```markdown
_I cannot italicize text that is exactly 80 characters long followed by 123456_, 123456.
```

`rumdl check` fails with:

```
example.md:1:81: [MD013] Line length 88 exceeds 80 characters
```

### Expected behavior

The line is broken after the comma, or no error is eported.

### Actual behavior

An error is reported.

### Configuration

```toml
[MD013]
line-length = 80
code-blocks = false
headings = false
reflow = true
reflow-mode = "normalize"
```

### Operating system

macOS

### Additional context

Both of these examples are formatted as expected (i.e., the line is broken):

```markdown
_I cannot italicize text that is exactly 80 characters long followed by 123456_, word.
```

```markdown
_I cannot italicize text that is exactly 80 characters long followed by 12345678_, 12345678.
```

**Comment by denialhaag** [comment]
The problem seems to have existed since version 0.2.33. The versions before break the line after the comma.

**Comment by rvben** [comment]
Fixed in v0.2.44.

Reflow refused to break a line before any `<up to 9 digits>.` or `<up to 9 digits>)` token, to avoid turning prose into an ordered list. That guard was too broad: a list can only interrupt a paragraph when it is numbered 1 and its first item has content, so a bare `123456.` is prose to the parser and wrapping before it cannot change the document. That is why `word.` and your other variants wrapped and this one did not, and why you got the unfixable "Line length" message instead of the fixable normalize one.

Your input now gives:

```markdown
_I cannot italicize text that is exactly 80 characters long followed by 123456_,
123456.
```

The regression tests use the parser as the oracle: they render the original paragraph and the reflowed output and require the two to match, so a wrap that changes the document's structure fails no matter which line it lands on.

Thanks for the bisect and for the working variants. Those controls are what showed the guard was firing on the wrong thing.
